libs/libc/math/lib_scalbnf.c

```diff
@@ -0,0 +1,76 @@
+/****************************************************************************
+ * libs/libc/math/lib_scalbnf.c
+ * get a double number of x*2^n
+ *
+ * This file is a part of NuttX:
+ *
+ *   Copyright (C) 2012 Gregory Nutt. All rights reserved.
+ *   Ported by: Darcy Gong
+ *
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <math.h>
+#include <stdint.h>
+
+#define FLT_MIN 0x1p-126f
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: scalbnf
+ *
+ * Description:
+ *   get a float number of x*2^n
+ *
+ ****************************************************************************/
+
+float scalbnf(float x, int n)
+{
+  union
+    {
+      float f;
+      uint32_t i;
+    }u;
+
+  float_t y = x;
+
+  if (n > 127)
+    {
+      y *= 0x1p127f;
+      n -= 127;
+      if (n > 127)
+        {
+          y *= 0x1p127f;
+          n -= 127;
+          if (n > 127)
+            {
+              n = 127;
+            }
+        }
+    }
+  else if (n < -126)
+    {
+      y *= FLT_MIN * 0x1p24f;
```

Comment:
Note that if the above suggestion is not committed, this reads as:

y *= 0x1p - (126f * 0x1p24f);

but if the above suggestion is committed, this reads as:

y *= (0x1p - 126f) * 0x1p24f;

tl;dr: Committing the above suggestion will change the outcome here. Is there a mistake somewhere?